pyam/core.py

```diff
@@ -992,23 +941,27 @@ def check_aggregate_region(self, variable, region='World', subregions=None,
             return
 
         df_region, df_subregions = (
-            _aggregate(self.data[rows], 'region')
+            groupby(self.data[rows], 'region')
             .align(df_subregions)
         )
 
         # use `np.isclose` for checking match
-        diff = df_region[~np.isclose(df_region, df_subregions, **kwargs)]
+        rows = ~np.isclose(df_region, df_subregions, **kwargs)
 
-        if len(diff):
-            msg = (
-                '`{}` - {} of {} rows are not aggregates of subregions'
-            )
-            logger.info(msg.format(variable, len(diff), len(df_region)))
+        # if region and subregions don't match, return inconsistent data
+        if sum(rows):
+            msg = '`{}` - {} of {} rows are not aggregates of subregions'
+            logger.info(msg.format(variable, sum(rows), len(df_region)))
 
             if exclude_on_fail:
-                self._exclude_on_fail(diff.index.droplevel([2, 3]))
+                self._exclude_on_fail(_meta_idx(df_region[rows].reset_index()))
 
-            return IamDataFrame(diff, region=region).timeseries()
+            _df = pd.concat(
+                [pd.concat([df_region[rows], df_subregions[rows]], axis=1,
+                           keys=(['region', 'subregions']))],
+                keys=['World'], names=['region'])
```

Comment:
```suggestion
                keys=[region], names=['subregions'])
```

Or do I misunderstand what is going on here (hard-coding 'World' doesn't look right)?

Reply:
the first part of this is indeed an error (it should use the kwarg `region`), but the name of that index level should be `region`